lab_1_classify_by_unigrams/main.py

```diff
@@ -85,6 +128,20 @@ def compare_profiles(
     'freq' in arguments, None is returned
     """
 
+    if not unknown_profile or not profile_to_compare:
+        return None
+    if (not isinstance(unknown_profile['freq'], dict)
+            and not isinstance(profile_to_compare['freq'], dict)):
+        return None
+    if not ('name' in unknown_profile and 'name' in profile_to_compare):
+        return None
+    if not ('freq' in unknown_profile and 'freq' in profile_to_compare):
+        return None
+    all_tokens = set(unknown_profile['freq'].keys()).union(set(profile_to_compare['freq'].keys()))
```

Comment:
can you explain what is going here?

Reply:
If the key 'freq' isnt in unknown_profile and in profile_to_compare None is returned

all_tokens is counting tokens by uniting two sets of keys (which are extracted from the dicts), making another set

unknown_profile is the dict where the key 'freq'  is stored, unknown_profile['freq'] is the needed token and .keys() extracts it